chat/src/main/java/com/github/twitch4j/chat/TwitchChat.java

```diff
@@ -327,6 +345,23 @@ public TwitchChat(EventManager eventManager, CredentialManager credentialManager
 
         // register event handler
         eventManager.onEvent(ChannelMessageEvent.class, this::onChannelMessage);
+        eventManager.onEvent(IRCMessageEvent.class, event -> {
+            // we get at least one room state event with channel name + id when we join a channel, so we cache that to provide channel id + name for all events
+            if ("ROOMSTATE".equalsIgnoreCase(event.getCommandType())) {
+                // check that channel id / name are present and that we didn't leave the channel yet
+                if (event.getChannelId() != null) {
+                    channelCacheLock.lock();
+                    try {
+                        event.getChannelName().map(String::toLowerCase).filter(currentChannels::contains).ifPresent(name -> {
+                            channelNameToChannelId.put(name, event.getChannelId());
+                            channelIdToChannelName.put(event.getChannelId(), name);
```

Comment:
If a name change has occurred, shall we remove the old name => id mapping?

Reply:
added `Optional.ofNullable(channelIdToChannelName.get(event.getChannelId())).ifPresent(oldName -> channelNameToChannelId.remove(oldName));` to account for name-changes, good point.

Reply:
Nice, it might be marginally safer to replace the ifPresent block with

```java
String oldName = channelIdToChannelName.put(event.getChannelId(), name);
if (!name.equals(oldName)) {
    channelNameToChannelId.remove(oldName, event.getChannelId());
    channelNameToChannelId.put(name, event.getChannelId());
}
```

as this avoids an id => name mapping being missing in the case of a concurrent read